helper/include/odp/helper/threads.h

```diff
@@ -86,13 +94,127 @@ typedef struct {
 			int		status;	/**< Process state chge status*/
 		} proc;
 	};
-} odph_odpthread_t;
+} odph_thread_t;
 
 /** Linux helper options */
 typedef struct {
 	odp_mem_model_t mem_model; /**< Process or thread */
 } odph_helper_options_t;
 
+/** Legacy thread table entry */
+typedef odph_thread_t odph_odpthread_t;
+
+/** Legacy thread parameters */
+typedef odph_thread_param_t odph_odpthread_params_t;
+
+/** Common parameters for odph_thread_create() call */
+typedef struct {
+	/**
+	 * ODP instance handle
+	 *
+	 * This is used for all threads, instead of 'instance' field of per
+	 * thread parameters (odph_thread_param_t).
+	 */
+	odp_instance_t instance;
+
+	/**
+	 * CPU mask for thread pinning
+	 */
+	const odp_cpumask_t *cpumask;
+
+	/**
+	 * Select between Linux pthreads and processes
+	 *
+	 * 0: Use pthreads
+	 * 1: Use processes
+	 *
+	 * Default value is 0.
+	 */
+	int thread_model;
+
+	/**
+	 * Synchronized thread creation
+	 *
+	 * 0: Don't synchronize thread creation
+	 * 1: Create threads in series so that the next thread is created
+	 *    only after the previous thread have signaled that it has passed
+	 *    ODP local initialization.
+	 *
+	 * Default value is 0.
+	 */
+	int sync;
+
+	/**
+	 * Thread parameter sharing
+	 *
+	 * 0: Thread parameters are not shared. The thread parameter table
+	 *    contains 'num' elements.
+	 * 1: The thread parameter table contains a single element, which is
+	 *    used for creating all 'num' threads.
+	 *
+	 * Default value is 0.
+	 */
+	int share_param;
+
+} odph_thread_common_param_t;
+
+/**
+ * Create and pin threads (as Linux pthreads or processes)
+ *
+ * This is an updated version of odph_odpthreads_create() call. It may be called
+ * multiple times to create threads in steps. Each call launches 'num' threads
+ * and pins those to separate CPUs based on the cpumask. Use 'thread_model'
+ * parameter to select if Linux pthreads or processes are used. This selection
+ * may be overridden with ODP helper options. See e.g. --odph_proc under
+ * odph_options() documentation.
+ *
+ * Thread creation may be synchronized by setting 'sync' parameter. It
+ * serializes thread start up (odp_init_local() calls), which helps to
+ * stabilize application start up sequence.
+ *
+ * By default, the thread parameter table contains 'num' elements, one for
+ * each thread to be created. However, all threads may be created
+ * with the same parameters by setting 'share_param' parameter.
```

Comment:
It is a little unclear how many elements should `thr_param[]` have in `share_param` case. I presume the answer is 1.

Reply:
Fixed in v2. It was documented in 'share_param' documentation, but now also here.